watch/funcs.go

```diff
@@ -41,7 +42,7 @@ func keyWatch(params map[string]interface{}) (WatcherFunc, error) {
 	}
 	fn := func(p *Plan) (uint64, interface{}, error) {
 		kv := p.client.KV()
-		opts := consulapi.QueryOptions{AllowStale: stale, WaitIndex: p.lastIndex}
+		opts := makeQueryOptionsWithContext(p, stale)
```

Comment:
The doc says that it should do that, will fix. [Doc](https://godoc.org/context) also said that go vet should catch it, but it didn't - _The go vet tool checks that CancelFuncs are used on all control-flow paths._